pkg/dependencies/factory.go

```diff
@@ -253,7 +253,7 @@ func (f *Factory) UseExecutablesDockerClient(client executables.DockerClient) *F
 
 // dockerLogin performs a docker login with the ENV VARS.
 func dockerLogin(ctx context.Context, registry string, docker executables.DockerClient) error {
-	username, password, _ := config.ReadCredentials()
+	username, password, _ := registrymirror.Credentials()
```

Comment:
Whats the rationale for using the `registrymirror` pkg for reading the credentials? The credentials read from that func seem like generic registry creds as opposed to being specific to a mirror.

Reply:
As per our doc says (https://anywhere.eks.amazonaws.com/docs/reference/clusterspec/optional/registrymirror/), `REGISTRY_USERNAME` and `REGISTRY_PASSWORD` are used for registry mirror authentication. Given that context, it makes more sense to me to move that to `registrymirror` pkg.

Reply:
Thanks Vincent for the clarification. As Guillermo says though, its still good practice to isolate reading configuration because it means we can change that source without touching lots of packages.